### PR TITLE
Add `branch_filter` to hooks API endpoints

### LINKED ISSUE #26591: REST API does not return branch filter for
### Description

As can be seen in the Swagger documentation for the repoListHooks and repoGetHook API calls, the branch filter is not returned when listing or getting webhooks.

https://try.gitea.io/api/swagger#/repository/repoListHooks
https://try.gitea.io/api/swagger#/repository/repoGetHook

It is possible to create or edit a webhook with a branch filter though.

https://try.gitea.io/api/swagger#/repository/repoCreateHook
https://try.gitea.io/api/swagger#/repository/repoEditHook

### Gitea Version

1.20.2

### Can you reproduce the bug on the Gitea demo site?

Yes

### Log Gist

_No response_

### Screenshots

_No response_

### Git Version

_No response_

### Operating System

_No response_

### How are you running Gitea?

Docker

### Database

PostgreSQL

### PR DESCRIPTION
**Pull request by yardenshoham**
We now include the branch filler in the response.

- Closes #26591 

# Before
image

# After
image

### CODE REVIEW

File: modules/structs/hook.go [hunk 1]
```diff
@@ -19,6 +19,7 @@ var ErrInvalidReceiveHook = errors.New("Invalid JSON payload received over webho
 type Hook struct {
 	ID                  int64             `json:"id"`
 	Type                string            `json:"type"`
+	BranchFilter        string            `json:"branch_filter" binding:"GlobPattern"`
```
Comment (wxiaoguang) [inline modules/structs/hook.go]: What is `binding:"GlobPattern"`?

Comment (yardenshoham) [reply modules/structs/hook.go]: I don't know, I just copied from https://github.com/go-gitea/gitea/blob/79c3329502217b664ef81d3f7f65aaf58282d29d/modules/structs/hook.go#L48

**Review by silverwind — APPROVED** [review]

**Review by wxiaoguang — CHANGES_REQUESTED** [review]
I do not think it's right to merge "don't know" code into code base. Every line of code should have clear meaning.

**Comment by yardenshoham** [comment]
I removed the unknown code

**Review by wxiaoguang — APPROVED** [review]
Actually I would suggest to figure out every line's detail by some in-depth research when a question comes, it makes people more familiar with the code base and framework 😄 

The `binding` tag is a framework mechanism, to help to "bind" user input to a value field, so it is only needed when using the binding framework to process user inputs.

**Pull request merged**